### ISSUE TITLE
[Bug][frontend] maxpool2d have a inconsistent inference results with PyTorch

### ISSUE THREAD

**Issue by jikechao** [apache/tvm#15281]
For the layer instance, TVM gives different results with Pytorch.

torch.nn.functional.max_pool2d(torch.randn([7, 175, 5, 3]), kernel_size=3,padding=1,ceil_mode=True)

### Actual behavior

![image](https://github.com/apache/tvm/assets/29506758/3511b47f-25e9-4234-ba89-d493e326dc9e)

### Steps to reproduce

```
import torch
from tvm import relay
import tvm
import numpy as np
from torch.nn import Module

input_data = torch.randn([7, 175, 5, 3], dtype=torch.float32)
class max_pool2d(Module):
    def forward(self, *args):
        return torch.nn.functional.max_pool2d(args[0], kernel_size=3,padding=1,ceil_mode=True)

m = max_pool2d().float().eval()

torch_outputs = m(input_data)

trace = torch.jit.trace(m, input_data)
input_shapes = [('input0', torch.Size([7, 175, 5, 3]))]

mod, params = relay.frontend.from_pytorch(trace, input_shapes)
with tvm.transform.PassContext(opt_level=3):
    exe = relay.create_executor('graph', mod=mod, params=params, device=tvm.device('llvm', 0), target='llvm').evaluate()

input_tvm = {'input0': np.array(input_data, dtype='float32')}
tvm_outputs = exe(**input_tvm).asnumpy()

np.testing.assert_allclose(torch_outputs, tvm_outputs, rtol=1e-3, atol=1e-3)

```

### Triage

* needs-triage
* frontend:pytorch
